### DISCUSSION TITLE
Inovelli Blue 2-1 Switch Latest Production Firmware Upgrade

### DISCUSSION THREAD

**Discussion by switzer60** [Koenkk/zigbee2mqtt#22474] [Devices]
First, thank you to all the folks that have put in the time to make this wonderful product.  It's been a godsend in terms of having a much more reliable experience with my Zigbee devices, and I love the ease of integration with MQTT!

Thank you!

I have 2 of the above devices.  I'm speculating that I updated on to version 2.18 just before: 
https://github.com/Koenkk/zigbee-OTA/pull/488 which set the version to 2.15.  I'd like to update the other device to firmware version 2.18

I'm running zigbee2mqtt in a docker container.
Zigbee2MQTT version: [1.36.0] commit: [86ed71c]
Coordinator type: EZSP v12
Dongle: usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20231007150852-if00

I've read the instructions on how to manually override the firmware version here:
https://www.zigbee2mqtt.io/guide/usage/ota_updates.html

I've modified my configuration.yaml
ota:
  ikea_ota_use_test_url: true
  zigbee_ota_override_index_location: /app/data/my_index.json

I've copied index.json from here:
https://github.com/Koenkk/zigbee-OTA/blob/master/index.json

I've added this entry:
    {
        "manufacturerCode": 4655,
        "imageType": 257,
        "url": "/app/data/VZM31-SN_2.18-Beta.ota",
        "force": true
    }

root@6cc93168032a:/app/data# ls -ascl /app/data/
total 536
  4 drwxr-xr-x 3 root root   4096 May  6 02:17 .
  4 drwxr-xr-x 3 root root   4096 May  6 02:00 ..
268 -rw-r--r-- 1 root root 270651 May  6 01:58 VZM31-SN_2.18-Beta.ota
  4 -rw-r--r-- 1 root root   3166 May  6 01:43 configuration.yaml
  4 -rw-r--r-- 1 root root    781 May  6 02:00 coordinator_backup.json
 48 -rw-r--r-- 1 root root  47540 May  6 02:00 database.db
  4 drwxr-xr-x 2 root root   4096 May  6 02:03 log
188 -rw-r--r-- 1 root root 189777 May  6 01:58 my_index.json
 12 -rw-r--r-- 1 root root  10544 May  6 02:00 state.json

My log output looks like this:

debug 2024-05-05 21:59:38: OTA: Checking if an update is available for '0xb43a31fffe3843e0' (VZM31-SN)
debug 2024-05-05 21:59:38: OTA: Using endpoint '1'
debug 2024-05-05 21:59:40: OTA: Got request '{"fieldControl":0,"manufacturerCode":4655,"imageType":257,"fileVersion":16908818}'
debug 2024-05-05 21:59:40: OTA: Is new image available for '0xb43a31fffe3843e0' (VZM31-SN), current '{"fieldControl":0,"manufacturerCode":4655,"imageType":257,"fileVersion":16908818}'
debug 2024-05-05 21:59:40: InovelliOTA: call getImageMeta for VZM31-SN
debug 2024-05-05 21:59:40: OTA: Is new image available for '0xb43a31fffe3843e0' (VZM31-SN), latest meta '{"fileVersion":16908818,"url":"https://github.com/InovelliUSA/Firmware/raw/3a73f98512f6e6f44fd5c89b7ee3402be5bbc4d1/Blue-Series/Zigbee/VZM31-SN-2-1-Switch/Beta/2.18/VZM31-SN_2.18-Beta.ota"}'
debug 2024-05-05 21:59:40: OTA: Update available for '0xb43a31fffe3843e0' (VZM31-SN): NO

Oddly, it doesn't look like from the logs it's using the my_index.json.

How can I get this device updated to 2.18?  What am I doing wrong?

Appreciate the help!

SixO

**Comment by grraab2000** [comment]
Howdy,
Did you get that to work?

**Comment by switzer60** [comment]
I haven't had more time to work on it.

**Comment by grraab2000** [comment]
I did similar:


configuration.yaml
-------------
ota:
 ikea_ota_use_test_url: true
 zigbee_ota_override_index_location: /config/my_index.json

my_index.json
-----------
[
 {
   "fileSize": 310258,
   "manufacturerCode": 4655,
   "imageType": 257,
   "sha512":
"6d480a5d621a16bb3a57fcc1af09071fc528dab2a8f3e479620c3bd75ddfa9e8f624c32b1dc35d5c1bc8db0f67a70ee150ce3a516a2026f717076dcbeba23df7",

   "url": "
https://github.com/InovelliUSA/Firmware/blob/main/Blue-Series/Zigbee/VZM31-SN-2-1-Switch/Beta/2.18/VZM31-SN_2.18-Beta.ota",

   "path": "/config/firmware/VZM31-SN_2.18-Beta.ota",
   "force": true
 },
]

After this, I had to force HA knowing it:
https://github.com/Koenkk/zigbee2mqtt/issues/14891#issuecomment-1312571066



G. Richard Raab

303.483.5720 (C)

***@***.***
